lib/charms/observability_libs/v1/cert_handler.py

```diff
@@ -87,15 +89,164 @@ class CertHandlerEvents(ObjectEvents):
     cert_changed = EventSource(CertChanged)
 
 
+class _VaultBackend(abc.ABC):
+    def store(self, contents: Dict[str, str], clear: bool = False): ...
+
+    def get_value(self, key: str) -> Optional[str]: ...
+
+    def retrieve(self) -> Dict[str, str]: ...
```

Comment:
If we manage just one secret (label) then this method is redundant.

Reply:
the difference between get value and retrieve is not the secret they retrieve from, but whether to retrieve the whole secret contents or just part of it. 

`get_value(value)` is shorthand for `retrieve().get(value)`

Reply:
Vault wraps a single secret